File: CRUD/C#/Aplicacion Web con Ajax/sistema/sistema/Controllers/UsuariosController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using sistema.Models;
using sistema.Conexiones;
using sistema.Funciones;
using System.IO;

namespace sistema.Controllers
{
    public class UsuariosController : Controller
    {

        Conexiones.Conexion conexion_now = new Conexiones.Conexion();
        Funciones.Funciones funcion = new Funciones.Funciones();

        //

        public ActionResult Buscador()
        {
            if (Request.Cookies["uid"] != null)
            {
                if (conexion_now.valid_cookie(Request.Cookies["uid"].Value))
                {
                    if (conexion_now.valid_cookie_admin(Request.Cookies["uid"].Value))
                    {
                        return View("Buscador");
                    }
                    else
                    {
                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = "Acceso Denegado";
                        ViewBag.tipo = "error";
                        ViewBag.controlador = "Administracion";
                        ViewBag.accion = "Index";
                        ViewBag.Title = "Administracion";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                }
                else
                {
                    return RedirectToAction("LogOn", "Login");
                }
            }
            else
            {
                return RedirectToAction("LogOn", "Login");
            }
        }

        [ActionName("Buscador"), HttpPost]
        public ActionResult Buscador(BuscadorModel model)
        {
            if (Request.Cookies["uid"] != null)
            {
                if (conexion_now.valid_cookie(Request.Cookies["uid"].Value))
                {
                    if (conexion_now.valid_cookie_admin(Request.Cookies["uid"].Value))
                    {
                        ViewBag.buscador_activo = 1;
                        ViewBag.nombre_buscar = model.nombre_buscar;
                        ViewBag.Title = "Usuarios";
                        return View("Buscador");
                    }
                    else
                    {
                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = "Acceso Denegado";
                        ViewBag.tipo = "error";
                        ViewBag.controlador = "Administracion";
                        ViewBag.accion = "Index";
                        ViewBag.Title = "Administracion";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                }
                else
                {
                    return RedirectToAction("LogOn", "Login");
                }
            }
            else
            {
                return RedirectToAction("LogOn", "Login");
            }
        }

        public ActionResult Agregar()
        {
            if (Request.Cookies["uid"] != null)
            {
                if (conexion_now.valid_cookie(Request.Cookies["uid"].Value))
                {
                    if (conexion_now.valid_cookie_admin(Request.Cookies["uid"].Value))
                    {
                        return View("Agregar");
                    }
                    else
                    {
                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = "Acceso Denegado";
                        ViewBag.tipo = "error";
                        ViewBag.controlador = "Administracion";
                        ViewBag.accion = "Index";
                        ViewBag.Title = "Administracion";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                }
                else
                {
                    return RedirectToAction("LogOn", "Login");
                }
            }
            else
            {
                return RedirectToAction("LogOn", "Login");
            }
        }

        [ActionName("Agregar"), HttpPost]
        public ActionResult AgregarUsuario(UsuarioModel model)
        {
            if (Request.Cookies["uid"] != null)
            {
                if (conexion_now.valid_cookie(Request.Cookies["uid"].Value))
                {
                    if (conexion_now.valid_cookie_admin(Request.Cookies["uid"].Value))
                    {
                        string respuesta = "";
                        bool done = false;
                        if (ModelState.IsValid)
                        {
                            string nombre_usuario = model.nombre_usuario;
                            string clave = model.password;
                            int tipo = model.tipo;

                            if (nombre_usuario != "" && clave != "" && funcion.valid_number(tipo.ToString()))
                            {
                                if (conexion_now.comprobar_existencia_usuario_crear(nombre_usuario))
                                {
                                    respuesta = "El usuario " + nombre_usuario + " ya existe";
                                    done = false;
                                }
                                else
                                {
                                    string password_encoded = funcion.md5_encode(clave);
                                    if (conexion_now.agregarUsuario(nombre_usuario, password_encoded, tipo))
                                    {
                                        respuesta = "El usuario ha sido registrado exitosamente";
                                        done = true;
                                    }
                                    else
                                    {
                                        respuesta = "Ha ocurrido un error en la base de datos";
                                        done = false;
                                    }
                                }
                            }
                            else
                            {
                                respuesta = "Los datos ingresados en el formulario son invalidos";
                                done = false;
                            }

                        }
                        else
                        {
                            respuesta = "Los datos ingresados en el formulario son invalidos";
                            done = false;
                        }

                        string mensaje_tipo = "";

                        if (done == true)
                        {
                            mensaje_tipo = "success";
                        }
                        else
                        {
                            mensaje_tipo = "warning";
                        }

                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = respuesta;
                        ViewBag.tipo = mensaje_tipo;
                        ViewBag.Title = "Usuarios";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                    else
                    {
                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = "Acceso Denegado";
                        ViewBag.tipo = "error";
                        ViewBag.controlador = "Administracion";
                        ViewBag.accion = "Index";
                        ViewBag.Title = "Administracion";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                }
                else
                {
                    return RedirectToAction("LogOn", "Login");
                }
            }
            else
            {
                return RedirectToAction("LogOn", "Login");
            }

        }

        public ActionResult Editar(int id)
        {
            if (Request.Cookies["uid"] != null)
            {
                if (conexion_now.valid_cookie(Request.Cookies["uid"].Value))
                {
                    if (conexion_now.valid_cookie_admin(Request.Cookies["uid"].Value))
                    {
                        if (id > 0)
                        {
                            ViewBag.modoEditar = 1;
                        }
                        else
                        {
                            ViewBag.modoEditar = 0;
                        }
                        ViewBag.id_usuario = id;
                        ViewBag.Title = "Usuarios";
                        return View("Editar");
                    }
                    else
                    {
                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = "Acceso Denegado";
                        ViewBag.tipo = "error";
                        ViewBag.controlador = "Administracion";
                        ViewBag.accion = "Index";
                        ViewBag.Title = "Administracion";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                }
                else
                {
                    return RedirectToAction("LogOn", "Login");
                }
            }
            else
            {
                return RedirectToAction("LogOn", "Login");
            }
        }

        [ActionName("Editar"), HttpPost]
        public ActionResult Editar(AsignarUsuarioModel model)
        {
            if (Request.Cookies["uid"] != null)
            {
                if (conexion_now.valid_cookie(Request.Cookies["uid"].Value))
                {
                    if (conexion_now.valid_cookie_admin(Request.Cookies["uid"].Value))
                    {
                        string respuesta = "";
                        bool done = false;
                        if (ModelState.IsValid)
                        {
                            int id_usuario = model.id_usuario;
                            string nombre_usuario = model.nombre_usuario;
                            int tipo = model.tipo;
                            if (funcion.valid_number(id_usuario.ToString()) && nombre_usuario != "" && funcion.valid_number(tipo.ToString()))
                            {
                                if (conexion_now.asignar_usuario(id_usuario, tipo))
                                {
                                    respuesta = "El usuario ha sido asignado exitosamente";
                                    done = true;
                                }
                                else
                                {
                                    respuesta = "Ha ocurrido un error en la base de datos";
                                    done = false;
                                }
                            }
                            else
                            {
                                respuesta = "Los datos ingresados en el formulario son invalidos";
                                done = false;
                            }

                        }
                        else
                        {
                            respuesta = "Los datos ingresados en el formulario son invalidos";
                            done = false;
                        }

                        string mensaje_tipo = "";

                        if (done == true)
                        {
                            mensaje_tipo = "success";
                        }
                        else
                        {
                            mensaje_tipo = "warning";
                        }

                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = respuesta;
                        ViewBag.tipo = mensaje_tipo;
                        ViewBag.Title = "Usuarios";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                    else
                    {
                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = "Acceso Denegado";
                        ViewBag.tipo = "error";
                        ViewBag.controlador = "Administracion";
                        ViewBag.accion = "Index";
                        ViewBag.Title = "Administracion";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                }
                else
                {
                    return RedirectToAction("LogOn", "Login");
                }
            }
            else
            {
                return RedirectToAction("LogOn", "Login");
            }

        }

        public ActionResult Borrar(int id)
        {
            if (Request.Cookies["uid"] != null)
            {
                if (conexion_now.valid_cookie(Request.Cookies["uid"].Value))
                {
                    if (conexion_now.valid_cookie_admin(Request.Cookies["uid"].Value))
                    {
                        bool ok = false;
                        string respuesta = "";
                        if (funcion.valid_number(id.ToString()))
                        {
                            if (conexion_now.borrarUsuario(id))
                            {
                                respuesta = "El usuario ha sido borrado exitosamente";
                                ok = true;
                            }
                            else
                            {
                                respuesta = "Ha ocurrido un error en la base de datos";
                                ok = false;
                            }
                        }
                        else
                        {
                            respuesta = "ID invalido";
                            ok = false;
                        }
                        if (ok == true)
                        {
                            ViewBag.estado = 1;
                            ViewBag.titulo = "Usuarios";
                            ViewBag.contenido = "El usuario ha sido borrado exitosamente";
                            ViewBag.tipo = "success";
                            ViewBag.controlador = "Administracion";
                            ViewBag.accion = "Usuarios";
                            ViewBag.Title = "Usuarios";
                            return View("~/Views/Shared/_ViewMessage.cshtml");
                        }
                        else
                        {
                            ViewBag.estado = 1;
                            ViewBag.titulo = "Usuarios";
                            ViewBag.contenido = respuesta;
                            ViewBag.tipo = "warning";
                            ViewBag.Title = "Usuarios";
                            return View("~/Views/Shared/_ViewMessage.cshtml");
                        }
                    }
                    else
                    {
                        ViewBag.estado = 1;
                        ViewBag.titulo = "Usuarios";
                        ViewBag.contenido = "Acceso Denegado";
                        ViewBag.tipo = "error";
                        ViewBag.controlador = "Administracion";
                        ViewBag.accion = "Index";
                        ViewBag.Title = "Administracion";
                        return View("~/Views/Shared/_ViewMessage.cshtml");
                    }
                }
                else
                {
                    return RedirectToAction("LogOn", "Login");
                }
            }
            else
            {
                return RedirectToAction("LogOn", "Login");
            }
        }

	}
}

File: CRUD/Java usando NetBeans/Aplicacion Web usando MVC Spring V2/SistemaWebSpring/src/main/java/sistema/controllers/UsuariosController.java
// Written by Ismael Heredia in the year 2017
package sistema.controllers;

import java.util.ArrayList;
import javax.servlet.http.HttpServletRequest;
import javax.validation.Valid;
import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Controller;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.servlet.ModelAndView;
import sistema.bo.TipoUsuarioBO;
import sistema.bo.UsuarioBO;
import sistema.functions.Funciones;
import sistema.models.Buscador;
import sistema.models.Mensaje;
import sistema.models.AsignarUsuario;
import sistema.models.Usuario;

@Controller
public class UsuariosController {

    ApplicationContext context
            = new ClassPathXmlApplicationContext("Spring-Module.xml");

    UsuarioBO usuarioBO = (UsuarioBO) context.getBean("UsuarioBO");
    TipoUsuarioBO tipoUsuarioBO = (TipoUsuarioBO) context.getBean("TipoUsuarioBO");

    Funciones funcion = new Funciones();

    @ResponseStatus(value = HttpStatus.NOT_FOUND)
    public final class ResourceNotFoundException extends RuntimeException {
    }

    @RequestMapping("/administracion/usuarios")
    public ModelAndView UsuariosForm(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String usuario_logeado = funcion.get_username_cookie(request);
                Buscador buscador = new Buscador();
                int cantidad = usuarioBO.listUsuarios("").size();
                ArrayList usuarios = usuarioBO.listUsuarios("");
                ModelAndView model = new ModelAndView("usuarios/listar");
                model.addObject("usuario_logeado", usuario_logeado);
                model.addObject("Buscador", buscador);
                model.addObject("usuarios", usuarios);
                model.addObject("cantidad", cantidad);
                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping("/administracion/usuarios/listar")
    public ModelAndView FormListarProductosGET(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                return new ModelAndView("redirect:/administracion/usuarios");
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/listar", method = RequestMethod.POST)
    public ModelAndView FormListarUsuarios(@ModelAttribute("Buscador") Buscador datos, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String usuario_logeado = funcion.get_username_cookie(request);
                Buscador buscador = new Buscador();
                String nombre_buscar = datos.getNombre_buscar();
                ArrayList usuarios = usuarioBO.listUsuarios(nombre_buscar);
                int cantidad = usuarios.size();
                ModelAndView model = new ModelAndView("usuarios/listar");
                model.addObject("usuario_logeado", usuario_logeado);
                model.addObject("Buscador", buscador);
                model.addObject("usuarios", usuarios);
                model.addObject("cantidad", cantidad);
                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping("/administracion/usuarios/agregar")
    public ModelAndView AgregarForm(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String usuario_logeado = funcion.get_username_cookie(request);
                AsignarUsuario usuario = new AsignarUsuario();
                ArrayList tipos = tipoUsuarioBO.listTipoUsuario("");
                ModelAndView model = new ModelAndView("usuarios/agregar");
                model.addObject("usuario_logeado", usuario_logeado);
                model.addObject("Usuario", usuario);
                model.addObject("tipos", tipos);
                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/agregar", method = RequestMethod.POST)
    public ModelAndView FormAgregarUsuario(@Valid Usuario usuario, BindingResult result, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {

                String contenido = "";
                String tipo = "";

                if (result.hasErrors()) {
                    contenido = "Faltan datos";
                    tipo = "warning";
                } else {

                    String nombre = usuario.getNombre();
                    String clave = funcion.md5_encode(usuario.getClave());
                    int id_tipo = usuario.getId_tipo();
                    String fecha_registro = funcion.fecha_del_dia();
                    usuario.setClave(funcion.md5_encode(usuario.getClave()));
                    usuario.setFecha_registro(fecha_registro);

                    if (usuarioBO.check_exists_usuario_new(nombre)) {
                        contenido = "El usuario " + nombre + " ya existe";
                        tipo = "warning";
                    } else {
                        if (usuarioBO.insert(usuario)) {
                            contenido = "Usuario registrado";
                            tipo = "success";
                        } else {
                            contenido = "Ha ocurrido un error en la base de datos";
                            tipo = "error";
                        }
                    }
                }

                Mensaje mensaje = null;

                if ("success".equals(tipo)) {
                    mensaje = new Mensaje("Usuarios", contenido, tipo, "administracion/usuarios");
                } else {
                    mensaje = new Mensaje("Usuarios", contenido, tipo, "administracion/usuarios/agregar");
                }

                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);

                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }

        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/{id}/editar", method = RequestMethod.GET)
    public ModelAndView FormEditandoUsuario(@PathVariable int id, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String usuario_logeado = funcion.get_username_cookie(request);
                String contenido = "";
                String tipo = "";
                if (id > 0) {
                    Usuario usuario = usuarioBO.findByUsuarioId(id);
                    if (usuario == null) {
                        throw new ResourceNotFoundException();
                    } else {
                        ArrayList tipos = tipoUsuarioBO.listTipoUsuario("");
                        ModelAndView model = new ModelAndView("usuarios/editar");
                        model.addObject("usuario_logeado", usuario_logeado);
                        model.addObject("AsignarUsuario", usuario);
                        model.addObject("tipos", tipos);
                        return model;
                    }
                } else {
                    Mensaje mensaje = new Mensaje("Usuarios", "ID Inválido", "warning", "administracion/usuarios");
                    ModelAndView model = new ModelAndView("mensajes/redireccion");
                    model.addObject("mensaje", mensaje);
                    return model;
                }
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping("/administracion/usuarios/editar")
    public ModelAndView FormEditarUsuarioGET(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                return new ModelAndView("redirect:/administracion/usuarios");
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/editar", method = RequestMethod.POST)
    public ModelAndView FormEditarUsuario(@Valid AsignarUsuario usuario, BindingResult result, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {

                String contenido = "";
                String tipo = "";

                if (result.hasErrors()) {
                    contenido = "Faltan datos";
                    tipo = "warning";
                } else {

                    int id_usuario = usuario.getId();
                    String nombre = usuario.getNombre();
                    int id_tipo = usuario.getId_tipo();

                    if (usuarioBO.update(usuario)) {
                        contenido = "Usuario editado";
                        tipo = "success";
                    } else {
                        contenido = "Ha ocurrido un error en la base de datos";
                        tipo = "error";
                    }
                }

                Mensaje mensaje = null;

                if ("success".equals(tipo)) {
                    mensaje = new Mensaje("Usuarios", contenido, tipo, "administracion/usuarios");
                } else {
                    mensaje = new Mensaje("Usuarios", contenido, tipo, "administracion/usuarios/" + usuario.getId() + "/editar");
                }

                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);

                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }

        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/{id}/borrar", method = RequestMethod.GET)
    public ModelAndView FormConfirmarBorrarUsuario(@PathVariable int id, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String usuario_logeado = funcion.get_username_cookie(request);
                Usuario usuario = usuarioBO.findByUsuarioId(id);
                if (usuario == null) {
                    throw new ResourceNotFoundException();
                } else {
                    ModelAndView model = new ModelAndView("usuarios/borrar");
                    model.addObject("usuario_logeado", usuario_logeado);
                    model.addObject("Usuario", usuario);
                    return model;
                }
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/borrar", method = RequestMethod.POST)
    public ModelAndView FormBorrarUsuario(@ModelAttribute("Usuario") Usuario usuario, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String contenido = "";
                String tipo = "";
                int id = usuario.getId();
                if (id > 0) {
                    if (usuarioBO.delete(usuarioBO.findByUsuarioId(id)) == true) {
                        contenido = "Usuario borrado";
                        tipo = "success";
                    } else {
                        contenido = "Ha ocurrido un error en la base de datos";
                        tipo = "error";
                    }
                } else {
                    contenido = "ID Inválido";
                    tipo = "warning";
                }

                Mensaje mensaje = null;

                if ("success".equals(tipo)) {
                    mensaje = new Mensaje("Usuarios", contenido, tipo, "administracion/usuarios");
                } else {
                    mensaje = new Mensaje("Usuarios", contenido, tipo, "administracion/usuarios/" + usuario.getId() + "/borrar");
                }

                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);

                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping("/administracion/usuarios/borrar")
    public ModelAndView FormBorrarUsuarioGET(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                return new ModelAndView("redirect:/administracion/usuarios");
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("mensajes/redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

}
